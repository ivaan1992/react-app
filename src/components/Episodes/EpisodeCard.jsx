/*
import { React } from 'react';
import { useSelector } from 'react-redux';

const EpisodeCard = () => {
    
    const {} = useSelector((state) => state.selectedEpisode)
    
    return (
        <div className="episode-card">
            <h3 className="title"></h3>
            <p className="airdate"></p>
            <div className="character-container">

            </div>
        </div>
    )
}

export default EpisodeCard;  */
