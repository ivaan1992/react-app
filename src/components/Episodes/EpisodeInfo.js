import React from 'react';
import PropTypes from 'prop-types';

const EpisodeInfo = ({
  id,
}) => (
  <option
    type="button"
    value={id}
  >
    Episode
    {' '}
    {id.toString()}
  </option>
);

EpisodeInfo.propTypes = {
  id: PropTypes.number.isRequired,
};

export default EpisodeInfo;
