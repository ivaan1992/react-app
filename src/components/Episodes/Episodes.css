.d-flex {
  display: flex;
}

.episode-container {
  width: 100%;
  justify-content: space-between;
  margin: 0 auto;
  background-color: #24282f;
}

.episode-selector {
  width: 20%;
  padding-left: 1rem;
}

.selector {
  width: 80%;
  margin-right: 1rem;
}

.episode-info {
  flex-direction: column;
  text-align: center;
  margin-bottom: 2rem;
}

.episode-card-title-content {
  width: 80%;
  margin-right: 2rem;
  padding-bottom: 1rem;
}

.episode-info-title {
  color: #fa9805;
  font-size: 2.1875rem;
  padding-bottom: 0.0625rem;
}

.episode-info-date {
  font-size: 0.875rem;
  color: #1367bb;
  padding-bottom: 0.625rem;
}

.episode-info-number {
  font-size: 0.5rem;
}

.characters-container {
  border: 2px solid red;
}

.chars {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  gap: 3rem;
  width: 90%;
  margin: 0 auto;
}
