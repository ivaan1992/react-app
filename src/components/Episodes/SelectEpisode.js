import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { getEpisodes } from '../../redux/Episode/EpisodeRedux';

const SelectEpisode = ({ children, onChange }) => {
  const dispatch = useDispatch();

  useEffect(() => {
    getEpisodes({ dispatch });
  }, []);

  const handleChange = (e) => {
    onChange(e.target.value);
  };

  return (
    <select className="selector" name="episodes" onChange={handleChange}>
      {children}
    </select>
  );
};

SelectEpisode.propTypes = {
  children: PropTypes.node.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default SelectEpisode;
