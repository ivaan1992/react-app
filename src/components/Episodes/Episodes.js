import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getEpisodes, getEpisode } from '../../redux/Episode/EpisodeRedux';
import EpisodeInfo from './EpisodeInfo';
import SelectEpisode from './SelectEpisode';
import CharacterCatalogue from '../Characters/CharacterCatalogue';
import './Episodes.css';

const Episodes = () => {
  const dispatch = useDispatch();
  const total = useSelector((state) => state.episodeReducer.total);
  const selectedEpisode = useSelector((state) => state.episodeReducer.selectedEpisode);
  const chars = useSelector((state) => state.episodeReducer.characters);

  useEffect(() => {
    getEpisodes({ page: '1', dispatch });
  }, []);

  const setSelectedEpisode = (episode) => {
    getEpisode({ episodeParam: episode, dispatch });
  };

  return (
    <div className="d-flex episode-container">
      <div className="episode-selector">
        <SelectEpisode
          onChange={setSelectedEpisode}
        >
          {
            [...Array(total).keys()].map((id) => (
              <EpisodeInfo
                key={id + 1}
                id={id + 1}
              />
            ))
          }
        </SelectEpisode>
      </div>
      <div className="episode-card-title-content">
        <div className="d-flex episode-info">
          <h1 className="episode-info-title">{selectedEpisode.name}</h1>
          <h5 className="episode-info-date">{selectedEpisode.air_date}</h5>
          <h5 className="episode-info-number">{selectedEpisode.episode}</h5>
        </div>
        <div className="">
          <div className="chars">
            {chars.map(({
              image, id, location, name, status,
            }) => (
              <CharacterCatalogue
                img={image}
                key={id}
                location={location}
                name={name}
                status={status}
              />
            ))}
          </div>
        </div>
      </div>
    </div>
  );
};

export default Episodes;
