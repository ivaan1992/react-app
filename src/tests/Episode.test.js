import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Episodes from '../components/Episodes/Episodes';

it('Testing Episode', () => {
  const list = renderer
    .create(
      <Provider store={store}>
        <Episodes />
      </Provider>,
    )
    .toJSON();
  expect(list).toMatchSnapshot();
});
