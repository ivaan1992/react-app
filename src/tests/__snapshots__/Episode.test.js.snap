// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Episode 1`] = `
<div
  className="d-flex episode-container"
>
  <div
    className="episode-selector"
  >
    <select
      className="selector"
      name="episodes"
      onChange={[Function]}
    />
  </div>
  <div
    className="episode-card-title-content"
  >
    <div
      className="d-flex episode-info"
    >
      <h1
        className="episode-info-title"
      />
      <h5
        className="episode-info-date"
      />
      <h5
        className="episode-info-number"
      />
    </div>
    <div
      className=""
    >
      <div
        className="chars"
      />
    </div>
  </div>
</div>
`;
